### ISSUE TITLE
noUnusedLocals should be reported as a warning, not an error

### ISSUE THREAD

**Issue by octogonz** [microsoft/TypeScript#40058]
## Search Terms

TS6133 6133 noUnusedLocals 

This is related but more general: https://github.com/microsoft/TypeScript/issues/13408

## Suggestion

This `ts.Diagnostic` message is classified as an `Error` not a warning:
```
    "'{0}' is declared but its value is never read.": {
        "category": "Error",
        "code": 6133,
        "reportsUnnecessary": true
    },
```

## Examples

Suppose someone is writing new code and not quite finished yet.
```ts
const x: number = 123;
const y: number = 321;
const product: number = x*y;

// TODO: Do something with "product"
```
They have not used `x` yet, but they certainly plan to use it when they are done writing their code.  When you run `tsc` the output looks like this:

```
src/index.ts:9:5 - error TS6133: 'x' is declared but its value is never read.
```

With a professional build orchestrator, this "compiler error" will:

- be printed in bright red, distracting attention from legitimate errors (e.g. calling a function that is not defined)
- prevent downstream dependency projects from building
- possibly also prevent other tasks (e.g. Jest) from running for the current project (we don't run tests if the code doesn't compile)

Proposal: Make it a warning instead:
```
    "'{0}' is declared but its value is never read.": {
        "category": "Warning",
        "code": 6133,
        "reportsUnnecessary": true
    },
```

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
This is entirely predicated on #13408, so should just be tracked there.
